.clangd

```diff
@@ -1,2 +1,32 @@
 CompileFlags:
-  CompilationDatabase: build/       # Search build/ directory for compile_commands.json
+  # Search build/ directory for compile_commands.json
+  CompilationDatabase: build
+Diagnostics:
+  # Additional checks that are useful but too noisy to be included in .clang-tidy
+  ClangTidy:
+    Add:
+      - google-*
+      - misc-*
+      - performance-*
+      - readability-*
+    Remove:
+      - misc-no-recursion
+      - misc-unused-parameters
+      - readability-function-*
+      - readability-identifier-length
+      - readability-magic-numbers
+      - readability-redundant-function-ptr-dereference
```

Comment:
I still think the warnings are too noisy. I propose we add the following warnings as well.

```suggestion
      - readability-redundant-function-ptr-dereference
      - readability-isolate-declaration
      - bugprone-assignment-in-if-condition
      - readability-suspicious-call-argument
      - performance-no-int-to-ptr
      - readability-else-after-return
      - google-readability-function-size
      - cert-err33-c
```

Reply:
- `bugprone-assignment-in-if-condition` how often does this even come up and it's not a bug? unless it's some macro
- `cert-err33-c` is pretty serious, see [ERR33-C. Detect and handle standard library errors](https://wiki.sei.cmu.edu/confluence/display/c/ERR33-C.+Detect+and+handle+standard+library+errors)

the rest are style guide related, so I don't mind removing them, although we could only enable them for certain folders/files if they're too noisy for upstream vim files

Reply:
> how often does this even come up and it's not a bug?

Literally the entire codebase. Did you not test the PR lol?

> cert-err33-c is pretty serious, see [ERR33-C. Detect and handle standard library errors](https://wiki.sei.cmu.edu/confluence/display/c/ERR33-C.+Detect+and+handle+standard+library+errors)

Same thing, it's a valid but it will create massive amounts of noise. This will just result in all of the functions getting a `(void)` slapped on them instead of any error checking.

Reply:
As discussed, this was a a new check that's only available in v15+ which is why I didn't notice it. I've disabled both for now.